src/eg/src/zk.rs

```diff
@@ -150,6 +153,15 @@ impl ProofRange {
 
         let mut w = <Vec<BigUint>>::with_capacity(big_l + 1);
         for j in 0..big_l + 1 {
+            let c_j_at_least_0 = self.0[j].c >= BigUint::from(0u8);
+            let c_j_less_than_2_256 = self.0[j].c.bits() <= 256; // c_j < 2^256
+            let v_j_at_least_0 = self.0[j].v >= BigUint::from(0u8);
```

Comment:
Does this do anything? Since its Uint isn't it >= 0 by construction.

Reply:
Agree. It is redundant, but put it there anyway. Will remove.